sonarts-core/tests/ruling/snapshots/no-return-promise-inside-try-block

```diff
@@ -0,0 +1,15 @@
+src/angular/packages/core/src/application_ref.ts: 238
+src/angular/tools/@angular/tsc-wrapped/src/main.ts: 116
+src/vscode/src/vs/base/common/worker/simpleWorker.ts: 329
+src/vscode/src/vs/base/parts/tree/browser/treeModel.ts: 845
+src/vscode/src/vs/code/node/cliProcessMain.ts: 107
```

Comment:
it looks like they try to catch exception from `JSON.parse`, so I would say it's FP

Reply:
Better to use Promise.try in this cases